backend/data/src/main/kotlin/io/tolgee/service/translation/TranslationService.kt

```diff
@@ -372,9 +372,9 @@ class TranslationService(
   }
 
   @Transactional
-  fun setState(keyIds: List<Long>, languageIds: List<Long>, state: TranslationState) {
+  fun setStateBatch(keyIds: List<Long>, languageIds: List<Long>, state: TranslationState) {
     val translations = getTargetTranslations(keyIds, languageIds)
-    translations.forEach { it.state = state }
+    translations.filter { it.state !== TranslationState.DISABLED }.forEach { it.state = state }
```

Comment:
The filtering can be done by database. Also !== is identity check. In this case it will work, but normaly it would be source of bugs.